src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,217 @@
+package duke.util;
+
+import java.util.regex.Pattern;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnknownCommand;
+import duke.exception.DukeException;
+
+/**
+ * Represents a parser to process and parse command.
+ */
+public class Parser {
+    /**
+     * Sets the possible types of input.
+     */
+    public enum InputType {
+        LIST,
+        MARK,
+        UNMARK,
+        DELETE,
+        BYE,
+        TODO,
+        DEADLINE,
+        EVENT,
+        FIND,
+        UNKNOWN
+    }
+
+    /**
+     * Returns the type of command by checking the keyword used at start of input.
+     *
+     * @param input Command input by user.
+     * @return Corresponding enum value of command.
+     */
+    public static InputType getCommandType(String input) {
+        if (input.equals("list")) {
+            return InputType.LIST;
+        } else if (input.startsWith("mark")) {
+            return InputType.MARK;
+        } else if (input.startsWith("unmark")) {
+            return InputType.UNMARK;
+        } else if (input.startsWith("delete")) {
+            return InputType.DELETE;
+        } else if (input.equals("bye")) {
+            return InputType.BYE;
+        } else if (input.startsWith("todo")) {
+            return InputType.TODO;
+        } else if (input.startsWith("deadline")) {
+            return InputType.DEADLINE;
+        } else if (input.startsWith("event")) {
+            return InputType.EVENT;
+        } else if (input.startsWith("find")) {
+            return InputType.FIND;
+        } else {
+            return InputType.UNKNOWN;
+        }
+    }
+
+    /**
+     * Checks if input matches the regex pattern.
+     *
+     * @param input Command entered by user.
+     * @param pattern Regex pattern of a valid command format.
+     * @return True if matches, false otherwise.
+     */
+    public static boolean matchPattern(String input, String pattern) {
+        Pattern regexPattern = Pattern.compile(pattern);
+        return regexPattern.matcher(input).matches();
+    }
```

Comment:
Since the compiled `regexPattern` is not used, you could do
```java
boolean b = Pattern.matches("a*b", "aaaaab");
```
instead, as shown in the [docs](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html)